### DISCUSSION TITLE
Is it possible to chain a query and a mutation into the same GraphQL call?

### DISCUSSION THREAD

**Discussion by nbouvrette** [hasura/graphql-engine#9260] [Question] [answered]
Let's say I want to search for a user with a specific email and if this user exists, create a token assigned to this user id - is this feasible in a single request, or do I need to make 2 requests (one query, one mutation)?

I couldn't find any example of this type of scenario anywhere.

**Comment by adas98012** [comment]
@nbouvrette 
You can use Hasura Actions for this use case. Here is the [reference doc](https://hasura.io/docs/latest/actions/index/#what-are-hasura-actions)

**Reply by nbouvrette** [reply]
Thanks @adas98012 - I saw the `UserLogin` example from the doc but unless I am misunderstanding, this would create the following circular sequence for us which would not be desirable:

- We call the action
- The action call back our API handler
- The API handler call back Hasura with a query to find the user
- The API handler calls Hasura again with a mutation to create a token
- The action returns the handler's output

This is actually 3 different GraphQL calls on top of more code to maintain which complexify the architecture. 

What I am looking for is the following sequence:

- We call Hasura with a Query + Mutation query to do a single transaction that:
  - Finds if a user exists
  - If it exists, do a mutation to create a token

Based on your answer I presume this is not doable today?

**Comment by adas98012** [comment]
@nbouvrette 
Thanks for the update.  Actually, in your case, the action will be a Hasura Query (to find the user) and the Token Creation will be part of the Handler (which will have the mutation that creates and returns the Token).

We have a previous [webinar](https://hasura.io/events/webinar/hasura-actions/?pg=docs&plcmt=body&cta=view-recording&tech=) that you may find informative.

Please let us know if you need any additional info.

**Comment by nbouvrette** [comment, marked as answer]
Yes, actually I asked [a similar question on Stackoverflow](https://stackoverflow.com/a/74693705/3764048) and I think I got the answer I was looking for:

**This is not possible today.**

Also, I was shared with this link which is an RFC being worked on to support that kind of feature: https://github.com/hasura/graphql-engine/issues/1573
